File: fontes/backend/java/fenix/src/main/java/com/t2ti/fenix/model/cadastros/PessoaAlteracao.java
/*******************************************************************************
Title: T2Ti ERP Fenix                                                                
Description: Model relacionado à tabela [PESSOA_ALTERACAO] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2020 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
package com.t2ti.fenix.model.cadastros;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.NamedQuery;
import javax.persistence.Table;
import java.util.Date;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

@Entity
@Table(name="PESSOA_ALTERACAO")
@NamedQuery(name="PessoaAlteracao.findAll", query="SELECT t FROM PessoaAlteracao t")
public class PessoaAlteracao implements Serializable {
	private static final long serialVersionUID = 1L;

    @Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;

    @Temporal(TemporalType.DATE)
	@Column(name="DATA_ALTERACAO")
	private Date dataAlteracao;

    @Column(name="OBJETO_ANTIGO")
	private String objetoAntigo;

    @Column(name="OBJETO_NOVO")
	private String objetoNovo;

    @ManyToOne
	@JoinColumn(name="ID_PESSOA")
    private Pessoa pessoa;

	public PessoaAlteracao() {
		//construtor padrão
	}

    public Integer getId() {
		return this.id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

    public Date getDataAlteracao() {
		return this.dataAlteracao;
	}

	public void setDataAlteracao(Date dataAlteracao) {
		this.dataAlteracao = dataAlteracao;
	}

    public String getObjetoAntigo() {
		return this.objetoAntigo;
	}

	public void setObjetoAntigo(String objetoAntigo) {
		this.objetoAntigo = objetoAntigo;
	}

    public String getObjetoNovo() {
		return this.objetoNovo;
	}

	public void setObjetoNovo(String objetoNovo) {
		this.objetoNovo = objetoNovo;
	}

    public Pessoa getPessoa() {
		return this.pessoa;
	}

	public void setPessoa(Pessoa pessoa) {
		this.pessoa = pessoa;
	}

		
}

File: fontes/backend/php/fenix/src/model/cadastros/PessoaAlteracao.php
<?php
/*******************************************************************************
Title: T2Ti ERP 3.0
Description: Model relacionado à tabela [PESSOA_ALTERACAO] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2021 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
declare(strict_types=1);

class PessoaAlteracao extends EloquentModel implements \JsonSerializable
{
    /**
     * The table associated with the model.
     *
     * @var string
     */
    protected $table = 'PESSOA_ALTERACAO';

    /**
     * Eager Loading - Relacionamentos que devem ser sempre carregados por padrão
     *
     * @var array
     */
	
    /**
     * Relations
     */

    /**
     * Gets e Sets
     */
    public function getIdAttribute() 
	{
		return $this->attributes['ID'];
	}

	public function setIdAttribute($id) 
	{
		$this->attributes['ID'] = $id;
	}

    public function getIdPessoaAttribute() 
	{
		return $this->attributes['ID_PESSOA'];
	}

	public function setIdPessoaAttribute($idPessoa) 
	{
		$this->attributes['ID_PESSOA'] = $idPessoa;
	}

    public function getDataAlteracaoAttribute() 
	{
		return $this->attributes['DATA_ALTERACAO'];
	}

	public function setDataAlteracaoAttribute($dataAlteracao) 
	{
		$this->attributes['DATA_ALTERACAO'] = $dataAlteracao;
	}

    public function getObjetoAntigoAttribute() 
	{
		return $this->attributes['OBJETO_ANTIGO'];
	}

	public function setObjetoAntigoAttribute($objetoAntigo) 
	{
		$this->attributes['OBJETO_ANTIGO'] = $objetoAntigo;
	}

    public function getObjetoNovoAttribute() 
	{
		return $this->attributes['OBJETO_NOVO'];
	}

	public function setObjetoNovoAttribute($objetoNovo) 
	{
		$this->attributes['OBJETO_NOVO'] = $objetoNovo;
	}


    /**
     * Mapping
     */
    public function mapear($objeto)
    {
        if (isset($objeto)) {
            isset($objeto->id) ? $this->setIdAttribute($objeto->id) : $this->setIdAttribute(null);

			$this->setIdPessoaAttribute($objeto->idPessoa);
			$this->setDataAlteracaoAttribute($objeto->dataAlteracao);
			$this->setObjetoAntigoAttribute($objeto->objetoAntigo);
			$this->setObjetoNovoAttribute($objeto->objetoNovo);

			// vincular objetos
        }
    }

    /**
     * Serialization
     * {@inheritdoc}
     */
    public function jsonSerialize()
    {
        return [
			'id' => $this->getIdAttribute(),
			'idPessoa' => $this->getIdPessoaAttribute(),
			'dataAlteracao' => $this->getDataAlteracaoAttribute(),
			'objetoAntigo' => $this->getObjetoAntigoAttribute(),
			'objetoNovo' => $this->getObjetoNovoAttribute(),
        ];
    }
}